src/Microsoft.AspNet.Hosting/HostingEngine.cs

```diff
@@ -252,6 +273,51 @@ public object GetService(Type serviceType)
             }
         }
 
+        // Represents an application environment that overrides the base path of the original
+        // application environment in order to make it point to the test application folder.
```

Comment:
comment is misleading. you're creating one of these for the mainline scenario as well, just with `appPath` and `appName` both `null`.